edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/PrometheusMetricsParser.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Newtonsoft.Json;
+
+    public static class PrometheusMetricsParser
+    {
+        // Extracts data from prometheous format. Read about prometheous format here:
+        // https://prometheus.io/docs/concepts/data_model/
+        const string PrometheusMetricSchema = @"^(?<metricname>[^#\{\}]+)(\{((?<tagname>[^="",]+)=(\""(?<tagvalue>[^="",]+)\"")(,(?<tagname>[^="",]+)=(\""(?<tagvalue>[^="",]+)\""))*)\})?\s(?<metricvalue>.+)$";
+        static readonly Regex PrometheusSchemaRegex = new Regex(PrometheusMetricSchema, RegexOptions.Compiled);
+
+        public static IEnumerable<Metric> ParseMetrics(DateTime timeGeneratedUtc, string prometheusMessage)
```

Comment:
how to check it is prometheuse message? what happen if non-prometheus message is given?

Reply:
the regex checks that. If it isn't a prometheous message, it returns an empty IEnumerable